codebase/src/java/disco/org/openlvc/disco/pdu/emissions/EmitterBeam.java

```diff
@@ -189,6 +262,31 @@ public final int getByteLength()
 	////////////////////////////////////////////////////////////////////////////////////////////
 	/// Helper Methods   ///////////////////////////////////////////////////////////////////////
 	////////////////////////////////////////////////////////////////////////////////////////////
+	/**
+	 * Returns the length of this structure in 32-bit words.
+	 * <p/>
+	 * If the 32-bit word length exceeds 255, then 0 is returned instead.
+	 * <p/>
+	 * This value is what is written to the data length field in the PDU. See DIS specification 
+	 * s7.6.2.
+	 * <p/>
+	 * For the length in bytes, please see {@link #getByteLength()}
+	 * 
+	 * @return the length of this structure in 32-bit words. If this figure would be greater than 
+	 *         255, then zero is returned
+	 */
+	public short getDataLength()
```

Comment:
Data length field calculation has now been moved into its own separate function and documented appropriately. Function also supports unit tests